sdk/src/types/block/output/delegation.rs

```diff
@@ -30,10 +30,13 @@ use crate::types::{
     ValidationParams,
 };
 
-impl_id!(pub DelegationId, 32, "Unique identifier of the Delegation Output, which is the BLAKE2b-256 hash of the Output ID that created it.");
-
-#[cfg(feature = "serde")]
-string_serde_impl!(DelegationId);
+crate::impl_id!(
+    /// Unique identifier of the [`DelegationOutput`](crate::types::block::output::DelegationOutput),
+    /// which is the BLAKE2b-256 hash of the [`OutputId`](crate::types::block::output::OutputId) that created it.
+    pub DelegationId {
+        pub const LENGTH: usize = 32;
```

Comment:
Why the whole thing and not just an integer?

Reply:
I think it better expresses what's happening 🤷‍♀️

Reply:
I would argue that we don't really need it to express anything, we're just generating code

Reply:
I like it this way and I don't think there's a good argument against it tbh

Reply:
All good, keep it. Just feel super random for no reason to me.